### ISSUE TITLE
Please document how to set up MAM

### ISSUE THREAD

**Issue by ssaavedra** [maranda/metronome#393]
Hi, I'm trying to set up a Metronome instance, but I can't find the way to correctly set up MAM in a way that will save my messages.

MAM support is advertised by the server, but does not store the messages.

I have a configuration such as:

```
modules_enabled = { "mam", "other modules" }

VirtualHost "jabber.test.dev"
        modules_enabled = { "mam", "other modules" }
        sql = { driver = "SQLite3", database = "/var/lib/metronome/metronome.sqlite" }
        storage = { mam = "sql", pep = "sql" }

        mam_save_time = 10
        mam_stores_cap = 1000000
        muc_log_mam_cache_expires = 1 * 30 * 24 * 3600 -- expire MUC chats after a month of inactivity

```

I have tried putting `mam` both in the global modules_enabled and in the VirtualHost one. I also tried both setting up the SQL driver (as configurations for Prosody recommend) and not setting it up. I also tried with `default_storage = "sql"` in case the key in the storage table was wrong in your documentation (which I think it is, according to the code, and should be "archiving" instead).
I also tried setting the mam_* variables globally instead of in the virtualhost, and I don't know what else to try :-)

A bit more documentation about this would be great.

**Comment by maranda** [comment]
Beside how debatable those configuration settings look… 1000000 messages cap in a store and SQLite will have your instance perform terribly even worse if you set mam_save_time to 10 seconds because Metronome will attempt to stash to disk every active archive which was changed after any 10 seconds, 5 minutes is more than appropriate. Also setting muc_log_mam_cache_expires that long doesn't make sense either.

I'm not sure what are you expecting, Metronome performs operations on the log caches, not on the DB itself so entries won't immediatly appear in the db, but only after either all resources on an account disconnect or mam_save_time is hit and log cache is changed, then it'll commit the archive to the store.

And SQL / SQLite3 are largely untested.

**Comment by ssaavedra** [comment]
Those settings were just for testing. mam_save_time was set to 10 seconds to ensure that the data was flushed to disk.

What settings are better tested then? I just tried sql because that's what works with Prosody, but it seems mod_mam has diverged a lot between both projects.

Both with sql and default (internal, I assume) storage systems nothing about MAM seens to get flushed to disk.

But persistence is not the only problem I encounter: I have tried to send+receive a message from one test account to another, and then from a different client (actually from jsxc through BOSH) issue a MAM request to get the history that should be in cache/ram at least, but the response is that no messages are stored on the server.
When querying the server, it returns that the test account has enabled MAM.

How can I further debug the issue, or which solution would you recommend me?

**Comment by maranda** [comment]
Ok I think I figured the problem here, the default prefs for MAM in Metronome are set to never log so, it won't do that as long as the user or the client tell it to do so explicitly, which is the correct behaviour and the one expected by clients. So you probably just need to set prefs for the account. 

I tested on my end and sqlite3 works, perhaps closing issue for now.
